modules/ovis/include/opencv2/ovis.hpp

```diff
@@ -164,6 +164,25 @@ class CV_EXPORTS_W WindowScene {
     CV_WRAP virtual void setEntityPose(const String& name, InputArray tvec = noArray(),
                                        InputArray rot = noArray(), bool invert = false) = 0;
 
+    /**
+     * update entity animation state
+     * @param name enitity name
+     * @param animname animation state name
+     * @param offset amount of time, in seconds, to extend the animation
+     */
+    CV_WRAP virtual void updateEntityAnimation(const String& name, const String& animname,
```

Comment:
this API as too low-level as it forces the user to use a timer to actually animate something. Instead automatically connect the animatonstate to the default ogre timer when the animation is enabled.
c.f.
https://github.com/OGRECave/ogre-meshviewer/blob/19.01/ogre_mesh_viewer.py#L229-L231

Reply:
Fully agreed and changed. Animation updates are now controlled by the frame controller.
To prevent multiple frame controllers for the same animation state to be created, and to allow us to clean up frame controllers, a new variable has been created (an unordered map bookkeeping over all created frame controllers).